packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -243,34 +271,47 @@ export function useAsyncList<T, C = string>(options: AsyncListOptions<T, C>): As
         selectedKeys: data.selectedKeys,
         sortDescriptor: action.sortDescriptor ?? data.sortDescriptor,
         signal: abortController.signal,
-        cursor: action.type === 'loadingMore' ? data.cursor : null
+        cursor: action.type === 'loadingMore' ? data.cursor : null,
+        filterText: data.filterText
       });
       dispatch({type: 'success', ...response, abortController});
     } catch (e) {
       dispatch({type: 'error', error: e, abortController});
     }
   };
 
+  let isInitialLoad = useRef(true);
+  // Handle initial load and reload on changes to filter text but only if performing server side filtering
+  // Kinda weird since it will trigger a reload when user selects a option
   useEffect(() => {
-    dispatchFetch({type: 'loading'}, load);
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+    if (isInitialLoad.current || !filterFn) {
+      dispatchFetch({type: 'loading'}, load);
+    }
+    isInitialLoad.current = false;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data.filterText, filterFn]);
```

Comment:
bit concerned that I couldn't include `load` in the dep array without running into an infinite loop. Not entirely sure what causes `load` to change, something to do with the dispatches in `dispatchFetch`

Reply:
maybe doesn't matter if we don't expect load to be changed by the user?